src/Mono.Android/Java.Interop/ManagedMarshalMethodsLookupTable.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Android.Runtime;
+
+namespace Java.Interop;
+
+internal class ManagedMarshalMethodsLookupTable
+{
+	[UnmanagedCallersOnly]
+	static unsafe void GetFunctionPointer (int assemblyIndex, int classIndex, int methodIndex, IntPtr target)
```

Comment:
Why `IntPtr target` and not `ref IntPtr target`?

Reply:
Because the C# compiler doesn't like that.  wat.

```
error CS8977: Cannot use 'ref', 'in', or 'out' in the signature of a method attributed with 'UnmanagedCallersOnly'
```

Reply:
But it should be possible to declare it as `IntPtr*` and avoid that cast later on 🤔 I will give that a try.